assets/src/js/homepage_transparent_nav.js

```diff
@@ -0,0 +1,45 @@
+export const setupTransparentNavHomepage = () => {
+  let isScrolled = false;
+
+  function ready(fn) {
+    if (document.readyState !== 'loading') {
+      fn();
+    } else {
+      document.addEventListener('DOMContentLoaded', fn);
+    }
+  }
+
+  function getYPosition() {
+    return (
+      window.pageYOffset ||
+    document.documentElement.scrollTop ||
+    document.body.scrollTop ||
+    0
+    );
+  }
+
+  function trackScrollAsBodyClass() {
+    const yTransitionPoint = 1; // in case people don't scroll ALL the way up
+    const yPosition = getYPosition();
+
+    if (yPosition > yTransitionPoint && !isScrolled) {
+      isScrolled = true;
+      document.body.classList.add('scrolled');
+    } else if (yPosition === 0 && isScrolled) {
+      isScrolled = false;
+      document.body.classList.remove('scrolled');
+    }
+  }
+
+  ready(() => {
+    if (!document.body.classList.contains('home')) {
```

Comment:
Maybe you can add a comment here to explain that this checks whether or not it's the homepage

Reply:
In fact, in general it would be nice to add JS docs to the new functions 🙂